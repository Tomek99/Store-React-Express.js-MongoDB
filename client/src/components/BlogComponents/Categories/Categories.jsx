import React from "react";
import styles from "./Categories.module.scss";
function Categories() {
  return (
    <div className={styles.Categories}>
      <h1 className={styles.categoriesHeader}>Categories</h1>
      <ul className={styles.categoriesList}>
        <li>Arabica</li>
        <li>Robusta</li>
        <li>Single Origin</li>
        <li>Blends</li>
        <li>Espresso</li>
        <li>Decaf</li>
        <li>Organic</li>
        <li>Flavored</li>
        <li>Cold Brew</li>
        <li>Instant</li>
      </ul>
    </div>
  );
}

export default Categories;
