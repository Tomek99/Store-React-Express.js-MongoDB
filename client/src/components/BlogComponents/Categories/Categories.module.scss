.Categories {
  color: var(--text-color);
  background-color: var(--light-creamy);
  padding: 20px 15px;
  border-radius: 12px;

  .categoriesHeader {
    font-size: 1.5rem;
    padding-bottom: 5px;
    margin-top: 10px;
    margin-bottom: 25px;
    border-bottom: var(--border);
  }

  .categoriesList {
    font-size: 1.3rem;
    list-style-type: none;
    display: flex;
    flex-direction: column;
    gap: 10px;

    li {
      align-self: flex-start;
      position: relative;
      transition: 0.3s linear;
      cursor: pointer;

      &::after {
        display: block;
        content: "";
        width: 0;
        height: 2px;
        background: var(--main-color);
        position: absolute;
        bottom: -3px;
        left: 0;
        transition: 0.3s linear;
      }

      &:hover::after {
        width: 100%;
      }

      &:hover {
        color: var(--main-color);
      }
    }
  }
}
