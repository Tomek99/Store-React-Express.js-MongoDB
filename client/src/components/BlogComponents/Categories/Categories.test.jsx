import { render } from "@testing-library/react";
import Categories from "./Categories";

test("should render component", () => {
  const { container } = render(<Categories />);

  expect(container).toMatchSnapshot();
});
