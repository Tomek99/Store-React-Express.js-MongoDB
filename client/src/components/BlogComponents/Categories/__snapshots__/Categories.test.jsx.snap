// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div>
  <div
    class="Categories"
  >
    <h1
      class="categoriesHeader"
    >
      Categories
    </h1>
    <ul
      class="categoriesList"
    >
      <li>
        Arabica
      </li>
      <li>
        Robusta
      </li>
      <li>
        Single Origin
      </li>
      <li>
        Blends
      </li>
      <li>
        Espresso
      </li>
      <li>
        Decaf
      </li>
      <li>
        Organic
      </li>
      <li>
        Flavored
      </li>
      <li>
        Cold Brew
      </li>
      <li>
        Instant
      </li>
    </ul>
  </div>
</div>
`;
